sway-core/src/control_flow_analysis/analyze_return_paths.rs

```diff
@@ -18,17 +18,30 @@ impl<'cfg> ControlFlowGraph<'cfg> {
     pub(crate) fn construct_return_path_graph<'eng: 'cfg>(
         engines: &'eng Engines,
         module_nodes: &[ty::TyAstNode],
-    ) -> Result<Self, CompileError> {
+    ) -> Result<Self, Vec<CompileError>> {
+        let mut errors = vec![];
+
         let mut graph = ControlFlowGraph::default();
         // do a depth first traversal and cover individual inner ast nodes
         let mut leaves = vec![];
         for ast_entrypoint in module_nodes {
-            let l_leaves = connect_node(engines, ast_entrypoint, &mut graph, &leaves)?;
-            if let NodeConnection::NextStep(nodes) = l_leaves {
-                leaves = nodes;
+            match connect_node(engines, ast_entrypoint, &mut graph, &leaves) {
+                Ok(l_leaves) => {
+                    if let NodeConnection::NextStep(nodes) = l_leaves {
+                        leaves = nodes;
+                    }
+                }
```

Comment:
```suggestion
                Ok(NodeConnection::NextStep(nodes)) => {
                    leaves = nodes;
                }
                Ok(_) => {}
```

Reply:
>  you shouldn't ever need to clear errors and warnings.
To recover, you don't need it. I will push improved errors on recovery for another PR, then.